### PR TITLE
client: add Tor support

### PR DESCRIPTION
**Pull request by dajohi**

### CODE REVIEW

File: client/cmd/dexc/ui/config.go [hunk 1]
```diff
@@ -72,6 +72,7 @@ type Config struct {
 	ReloadHTML bool   `long:"reload-html" description:"Reload the webserver's page template with every request. For development purposes."`
 	DebugLevel string `long:"log" description:"Logging level {trace, debug, info, warn, error, critical}"`
 	LocalLogs  bool   `long:"loglocal" description:"Use local time zone time stamps in log entries."`
+	TorProxy   string `long:"torproxy" description:"Connect via TOR (eg. 127.0.0.1:9050)."`
```
Comment (JoeGruffins) [inline client/cmd/dexc/ui/config.go:75]: Need to add TorProxyUsername and TorProxyPassword for them to show up on help `-h`, and also to be settable at all I think.

**Review by JoeGruffins — APPROVED** [review]
Working well. Tested with remote server and no problems trading. Just the the user:password not settable from config I believe.

File: client/cmd/dexc/ui/config.go (hunk 1 above)
Comment (chappjc) [reply client/cmd/dexc/ui/config.go:75]: I'm trying to wrap my mind around the use case for setting user/pass at dexc startup.  They dictate how the circuit is created, I believe.  If you set them, that makes tor use a specific circuit for dexc.  If not set, that uses whatever circuit it has? Or?

File: client/core/core.go [hunk 2]
```diff
@@ -3301,7 +3313,19 @@ func (c *Core) connectDEX(acctInfo *db.AccountInfo) (*dexConnection, error) {
...
 		},
 		Logger: c.log.SubLogger(wsURL.String()),
-	})
+	}
+	if c.cfg.TorProxy != "" {
+		proxy := &socks.Proxy{
+			Addr:         c.cfg.TorProxy,
+			Username:     c.cfg.TorProxyUsername,
+			Password:     c.cfg.TorProxyPassword,
+			TorIsolation: true,
```
Comment (chappjc) [inline client/core/core.go:3322]: Hmm, aren't user and pass ignored when TorIsolation=true?  https://github.com/decred/go-socks/blob/775b73941e73859df97ecc4f2f62952e754958a9/socks/dial.go#L120-L132

File: client/cmd/dexc/ui/config.go (hunk 1 above)
Comment (JoeGruffins) [reply client/cmd/dexc/ui/config.go:75]: I actually don't know how to set a password... I guess it is in the config file somewhere. May not be a useful setting?

File: client/core/core.go (hunk 2 above)
Comment (JoeGruffins) [reply client/core/core.go:3322]: Certainly looks that way.

File: client/cmd/dexc/ui/config.go (hunk 1 above)
Comment (chappjc) [reply client/cmd/dexc/ui/config.go:75]: Apparently it's not actually used by the tor socks5 server to authenticate, but to specify circuit uniqueness.  At least I think.  Input from dhill would be good here.

Comment (dajohi) [reply client/cmd/dexc/ui/config.go:75]: Essentially:  same user/pass = tor isolation off.  randomized user/pass each connection = tor isolation on.  or TorIsolation: yes causes go-socks to randomize the user/pass for you.

https://github.com/decred/go-socks/blob/master/socks/dial.go#L120

I will push a PR to remove User/Pass.. and just add a --torisolation bool.  Does that work?

File: client/core/core.go (hunk 2 above)
Comment (dajohi) [reply client/core/core.go]: removed user/pass and made --torisolation

**Review by chappjc — APPROVED** [review]
Ran with and without isolation to remote testnet server.  Even restarted the tor service a couple times while dexc was running.  All good.

**Pull request merged**